### DISCUSSION TITLE
Conditional mounts or mount after container creation

### DISCUSSION THREAD

**Discussion by Jogi7819** [devcontainers/community#56] [Q&A]
Hi community,

we use the same devcontainer.json file in different environments (remote and local builds). Some of the mounts are only required for a certain environment. 

1. Is it possible to make a mount optional (maybe based on environment variables) or to only have warning instead of an error if a certain mount point does not exists?

2. Is it possible to mount a folder on container creation for example when the initializeCommand stage is executed?

Thanks in advance

**Comment by bamurtaugh** [comment]
Thank you for opening @Jogi7819!

On the VS Code Dev Containers extension side, we have a feature request tracking conditional mounting: https://github.com/microsoft/vscode-remote-release/issues/5322.

As indicated in that issue (https://github.com/microsoft/vscode-remote-release/issues/5322#issuecomment-1339622701), there's an issue tracking this for the overall dev container spec here: https://github.com/devcontainers/spec/issues/132, and we'd certainly welcome any contributions to https://github.com/devcontainers/cli!
